components/camel-oaipmh/src/main/java/org/apache/camel/oaipmh/utils/OAIPMHHttpClient.java

```diff
@@ -0,0 +1,150 @@
+            }
+
+            if (token != null) {
+                builder.addParameter("resumptionToken", token);
+            } else {
+                if (metadataPrefix != null) {
+                    builder.addParameter("metadataPrefix", metadataPrefix);
+                }
+                if (set != null) {
+                    builder.addParameter("set", set);
+                }
+                if (from != null) {
+                    builder.addParameter("from", from);
+                }
+                if (until != null) {
+                    builder.addParameter("until", until);
+                }
+            }
+
+            HttpGet httpget = new HttpGet(builder.build());
+
+            RequestLine requestLine = httpget.getRequestLine();
+
+            LOG.info("Executing request: {} ", requestLine);
+
+            // Create a custom response handler
+            ResponseHandler<String> responseHandler = new ResponseHandler<String>() {
+
+                @Override
+                public String handleResponse(final HttpResponse response) throws IOException {
+                    int status = response.getStatusLine().getStatusCode();
+                    if (status >= 200 && status < 300) {
+                        HttpEntity entity = response.getEntity();
+                        if (entity == null) {
+                            throw new IOException("No response received");
+                        }
+                        return EntityUtils.toString(entity, Charset.forName("UTF-8"));
+                    } else {
+                        throw new ClientProtocolException("Unexpected response status: " + status);
+                    }
+                }
+
+            };
+            String responseBody = httpclient.execute(httpget, responseHandler);
+
+//            String uri = requestLine.getUri();
+//            System.out.println(uri);
+//            String sha256Hex = DigestUtils.sha256Hex(uri.split(".org")[1]);
+//            System.out.println ("File:"+sha256Hex);
+//            BufferedWriter writer = new BufferedWriter(new FileWriter("/tmp/tests/test4/" + sha256Hex + ".xml"));
+//            writer.write(responseBody);
+//            writer.close();
+            LOG.debug("Response received: {}", responseBody);
+
+            return responseBody;
+        } finally {
+            httpclient.close();
+        }
+    }
+
+    protected CloseableHttpClient getCloseableHttpClient() throws Exception {
+        if (isIgnoreSSLWarnings()) {
+            try {
+                SSLContextBuilder builder = new SSLContextBuilder();
+                builder.loadTrustMaterial(null, new TrustSelfSignedStrategy());
+                SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(
+                        builder.build());
+                return HttpClients.custom().setSSLSocketFactory(
+                        sslsf).build();
+            } catch (Exception ex) {
```

Comment:
If you are catching the exception and then throw it again, makes no sense. However, I don't feel comfortable of throwing generic `Exception` here, please be specific on which exception you want to throw here